src/toggle-button/__tests__/styles.test.tsx

```diff
@@ -0,0 +1,260 @@
+import { getToggleButtonStyles } from '../style';
+
+// Mock the environment module
+jest.mock('../../internal/environment', () => ({
+  SYSTEM: 'core',
+}));
+
+const STATES = ['default', 'disabled', 'hover', 'active', 'pressed'] as const;
+const STATE_PROPERTIES = ['background', 'borderColor', 'boxShadow', 'color'] as const;
+
+const CSS_PROPERTY_MAP = {
+  background: {
+    default: customCssProps.styleBackgroundDefault,
+    disabled: customCssProps.styleBackgroundDisabled,
+    hover: customCssProps.styleBackgroundHover,
+    active: customCssProps.styleBackgroundActive,
+    pressed: customCssProps.styleBackgroundPressed,
+  },
+  borderColor: {
+    default: customCssProps.styleBorderColorDefault,
+    disabled: customCssProps.styleBorderColorDisabled,
+    hover: customCssProps.styleBorderColorHover,
+    active: customCssProps.styleBorderColorActive,
+    pressed: customCssProps.styleBorderColorPressed,
+  },
+  boxShadow: {
+    default: customCssProps.styleBoxShadowDefault,
+    disabled: customCssProps.styleBoxShadowDisabled,
+    hover: customCssProps.styleBoxShadowHover,
+    active: customCssProps.styleBoxShadowActive,
+    pressed: customCssProps.styleBoxShadowPressed,
+  },
+  color: {
+    default: customCssProps.styleColorDefault,
+    disabled: customCssProps.styleColorDisabled,
+    hover: customCssProps.styleColorHover,
+    active: customCssProps.styleColorActive,
+    pressed: customCssProps.styleColorPressed,
+  },
+} as const;
+
+describe('getToggleButtonStyles', () => {
+  afterEach(() => {
+    jest.resetModules();
+  });
+
+  test('returns empty object for undefined or empty style objects', () => {
+    expect(getToggleButtonStyles(undefined)).toEqual({});
+    expect(getToggleButtonStyles({})).toEqual({});
+  });
+
+  test('handles all possible style configurations', () => {
+    const allStyles = {
+      root: {
+        borderRadius: '4px',
+        borderWidth: '1px',
+        paddingBlock: '8px',
+        paddingInline: '12px',
+        background: {
+          default: '#ffffff',
+          disabled: '#f0f0f0',
+          hover: '#fafafa',
+          active: '#eeeeee',
+          pressed: '#e0e0e0',
+        },
+        borderColor: {
+          default: '#cccccc',
+          disabled: '#e0e0e0',
+          hover: '#999999',
+          active: '#666666',
+          pressed: '#0073bb',
+        },
+        boxShadow: {
+          default: 'none',
+          disabled: 'none',
+          hover: '0 1px 2px rgba(0,0,0,0.1)',
+          active: '0 1px 2px rgba(0,0,0,0.2)',
+          pressed: '0 0 0 2px #0073bb',
+        },
+        color: {
+          default: '#000000',
+          disabled: '#999999',
+          hover: '#000000',
+          active: '#000000',
+          pressed: '#0073bb',
+        },
+        focusRing: {
+          borderColor: '#0073bb',
+          borderRadius: '6px',
+          borderWidth: '2px',
+        },
+      },
+    };
+
+    expect(getToggleButtonStyles(allStyles)).toEqual({
```

Comment:
If this test only validates the correct props assignment, why not use snapshot tests instead?